### PR TITLE
Use `Metadata-Flavor` when checking if Google metadata server is available

### PR DESCRIPTION
**Pull request by renchap**
The Google metadata server requires a `Metadata-Flaver: Google` header to bet set on every requests.

When you do not include it (like before in the `HEAD` request to check if it is available) then the metadata server returns a HTTP Forbidden error.

With the GCE metadata server, this HTTP response contains the `Metadata-Flavor` header.

But on Cloud Run, a different metadata server software is used and this server no longer returns the header on Forbidden responses.

This change makes Dregsy include the header on the `HEAD` request, so it can correctly detect the Google metadata server presence.
I also changed the function name to better reflect that it detects the Google Metadata server, not only in GCE.

### CODE REVIEW

**Comment by Klaitos** [comment]
I confirm this version works well on Cloud Run Jobs.
Without this PR, I observe `level=error msg="neither GOOGLE_APPLICATION_CREDENTIALS set, nor a GCE instance"`

File: internal/pkg/auth/gcr.go [hunk 1]
```diff
@@ -78,16 +74,24 @@ func (rf *gcrAuthRefresher) Refresh(creds *Credentials) error {
 	return nil
 }
 
-//
-func isGCEInstance() bool {
-	resp, err := http.Head(gcp_metadata_url)
+func hasGoogleMetadataServer() bool {
+	client := &http.Client{}
```
Comment (xelalexv) [inline internal/pkg/auth/gcr.go:78]: You could merge this into line 87.

**Review by xelalexv — COMMENTED** [review]
Thank you for this PR, and sorry for the late reply! This is looking good, just added one minor comment. ~~Plus a bit of nitpicking: Please keep the blank comment lines before function signatures, even if you don't like them ;-)~~ Just found out that starting with Go 1.19, `gofmt` removes these blank comments, so never mind.

**Comment by sanketg86** [comment]
Just curiosity, as I want to use dregsy to copy image to GAR.  so if this PR merge then it work in GKE env with workload identity ?

**Comment by renchap** [comment]
I just pushed a new commit with your requested changes.

The comments were indeed removed by `gofmt`, but I (manually) kept them in this PR, so it is consistent with the codebase. You may want to reformat everything in another PR to remove them in every files if you want :)

**Pull request merged**